### PR TITLE
[SLF4J-437] Add BOM module for better dependency management

### PR DESCRIPTION
**Pull request by Prototik**
BOM's used to keep in sync versions of dependencies. You can read more info [here](https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#Importing_Dependencies)

Example usage (Maven):
```xml
<dependencyManagement>
  <dependencies>
    <dependency>
      <groupId>org.slf4j</groupId>
      <artifactId>slf4j-bom</artifactId>
      <version>${slf4j.version}</version>
      <type>pom</type>
      <scope>import</scope>
    </dependency>
  </dependencies>
</dependencyManagement>
...
<dependencies>
  <dependency>
    <groupId>org.slf4j</groupId>
    <artifactId>slf4j-api</artifactId>
  </dependency>
  <dependency>
    <groupId>org.slf4j</groupId>
    <artifactId>slf4j-simple</artifactId>
  </dependency>
  <!-- whatever else -->
</dependencies>
```

Example usage (Gradle):
```gradle
dependencies {
  api(platform("org.slf4j:slf4j-bom:$slf4j_version"))

  api("org.slf4j:slf4j-api")
  api("org.slf4j:slf4j-simple")
}
```

### CODE REVIEW

File: slf4j-bom/pom.xml [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+      </dependency>
+      <dependency>
+        <groupId>${project.groupId}</groupId>
+        <artifactId>osgi-over-slf4j</artifactId>
+        <version>${slf4j.version}</version>
+      </dependency>
+    </dependencies>
+  </dependencyManagement>
+
+  <profiles>
```
Comment (reitzmichnicht) [inline slf4j-bom/pom.xml]: Any reason to keep this profile in the bom pom.xml?

**Comment by szymonk92** [comment]
Any plans for merge?

**Comment by AceHack** [comment]
Yes curious about merge too

**Comment by mikaelhg** [comment]
Bumped into this today, slf4j is the only BOM-less dependency in the project.

**Comment by JKomoroski** [comment]
Would love to see this.

**Comment by Prototik** [comment]
Implemented in 6fd9db9b5eef541d9648d486b5db096ed5d00b37

**Pull request closed without merging**